pkg/k8shandler/elasticsearch.go

```diff
@@ -440,6 +440,189 @@ func DoSynchronizedFlush(clusterName, namespace string, client client.Client) (b
 	return (payload.StatusCode == 200), payload.Error
 }
 
+// This will idempompotently update the index templates and update indices' replica count
+func UpdateReplicaCount(clusterName, namespace string, client client.Client, replicaCount int32) (bool, error) {
+
+	if ok, _ := updateAllIndexTemplateReplicas(clusterName, namespace, client, replicaCount); ok {
+		if ok, _ = updateAllIndexReplicas(clusterName, namespace, client, replicaCount); ok {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func updateAllIndexReplicas(clusterName, namespace string, client client.Client, replicaCount int32) (bool, error) {
+
+	indexHealth, _ := getIndexHealth(clusterName, namespace, client)
+
+	// get list of indices and call updateIndexReplicas for each one
+	for index, health := range indexHealth {
+		// only update replicas for indices that don't have same replica count
+		if parseInt32("replicas", health.(map[string]interface{})) != replicaCount {
+			// best effort initially?
+			logrus.Debugf("Updating %v from %d replicas to %d", index, parseInt32("replicas", health.(map[string]interface{})), replicaCount)
+			updateIndexReplicas(clusterName, namespace, client, index, replicaCount)
+		}
+	}
+
+	return true, nil
+}
+
+func getIndexHealth(clusterName, namespace string, client client.Client) (map[string]interface{}, error) {
+	payload := &esCurlStruct{
+		Method: http.MethodGet,
+		URI:    "_cat/indices?h=health,status,index,pri,rep",
+	}
```

Comment:
ok for now - but we need to use the JSON endpoints, or use the golang elasticsearch library - tech debt